README
Dist::Zilla::Plugin::Bootstrap::ShareDir::Module

This module allows one to load a "Module" styled "ShareDir" using a
"Bootstrap" mechanism so a distribution can use files in its own source
tree when building with itself.

This is very much like the "Bootstrap::lib" plugin in that it injects
libraries into @INC based on your existing source tree, or a previous
build you ran.

And it is syntactically like the "ModuleShareDirs" plugin.

Note that this is really only useful for self consuming *plugins* and
will have no effect on the "test" or "run" phases of your dist. ( For
that, you'll need "Test::File::ShareDir" ).

INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2015 by Kent Fredric
<kentfredric@gmail.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Release history for Dist-Zilla-Plugin-Bootstrap-ShareDir-Module

1.001003 2017-03-06T06:42:06Z c3c4d99

1.001002 2017-03-06T05:56:08Z c383b1c
 [Bugfix]
 - Avoid testing issues under -Ddefault_inc_excludes_dot
 - Closes https://github.com/kentnl/Dist-Zilla-Plugin-Bootstrap-ShareDir-Module/issues/1

 [Dependencies::Stats]
 - Dependencies changed since 1.001001, see misc/*.deps* for details
 - configure: ↓1
 - develop: +7 ↑4 -3 (suggests: ↑2)
 - runtime: ↓1 -2
 - test: ↓2

 [Internals]
 - Remove use of ConfigDumper

1.001001 2015-01-21T16:26:20Z af98f22
 [Dependencies::Stats]
 - Dependencies changed since 1.001000, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +10 ↑2 -2 (recommends: +1, suggests: ↑2)
 - runtime: -1 (recommends: +2)
 - test: +1 (recommends: ↑1 ↓1)

 [Documentation]
 - It has some now.

 [Internals]
 - Removed use of MooseX::AttributeShortcuts to remove indirect dependency on autobox.

1.001000 2014-08-31T14:00:46Z
 [00 Minor]
 - Tests refactored
 - Internals changed.
 - Metadata fixes.

 [Dependencies::Stats]
 - Dependencies changed since 1.000000, see misc/*.deps* for details
 - develop: +2 ↑1 (suggests: ↑2)
 - runtime: +1
 - test: +1 -2

 [Internals]
 - dump_config now implemented with ConfigDumper

 [Metadata]
 - author = kentnl

 [Tests]
 - Refactor all tests in terms of dztest util.

1.000000 2014-08-02T14:44:12Z
 [00 Major]
 - Changes that affect hand-operated downstream.

 [01 Toolkit Changes]
 - Switched to EUMM for install
 - "latest" dependencies now only recommended, not required.
 - https://questhub.io/realm/perl/quest/5388cc12bbd0be5b2500011f

 [02 Version Scheme]
 - Switched from x.y.z to x.yyyyyy
 - This is the primary reason for the 1.x on the box
 - https://questhub.io/realm/perl/quest/5388d3a9bbd0be2224000129

 [Dependencies::Stats]
 - Dependencies changed since 0.2.0, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +9 -11 (suggests: +2 -1)
 - runtime: +1 ↑1
 - test: +2 ↓1 -2 (recommends: +3)

 [Misc]
 - Path::FindDev no longer needlessly used for tests

0.2.0 2013-11-21T07:22:18Z
 [00 Minor]
 - Reimplement in terms of T:F:SD:O:M

 [Dependencies::Stats]
 - Dependencies changed since 0.1.3, see misc/*.deps* for details
 - runtime: +1 -1

0.1.3 2013-11-20T09:44:25Z
 [00 Minor]
 - Drop 5.10 requirement

 [Dependencies::Stats]
 - Dependencies changed since 0.1.2, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - develop: +43 ↑1 -1 (recommends: -1, suggests: +1 -1)
 - runtime: ↓1
 - test: ↑1 -3

 [Packaging]
 - Use new fallback stuff

 [Tests]
 - Regen ReportPrereqs to not report devel deps
 - Move from Test::Compile to Test::Compile::PerFile

0.1.2 2013-09-24T23:49:46Z
 [Bugfix]
 - use Test::DZil internally instead of calling dzil
 - log bootstrapping of modules instead of silence

 [Dependencies::Stats]
 - Dependencies changed since 0.1.1, see misc/*.deps* for details
 - test: +3

0.1.1 2013-09-08T11:23:50Z
 [Bugfix]
 - Need Bootstrap::lib for tests

 [Dependencies::Stats]
 - Dependencies changed since 0.1.0, see misc/*.deps* for details
 - test: +1

 [Docs]
 - Add metapod data.

0.1.0 2013-09-04T22:01:12Z
 - First version.


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.009.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Use a share directory on your dist for a module during bootstrap",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Bootstrap-ShareDir-Module",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::Bootstrap::ShareDir::Module",
  "PREREQ_PM" => {
    "Dist::Zilla::Role::Bootstrap" => 0,
    "Moose" => 0,
    "Path::Tiny" => 0,
    "Test::File::ShareDir::Object::Module" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Dist::Zilla::Plugin::Bootstrap::lib" => "0.04000000",
    "Dist::Zilla::Util::Test::KENTNL" => "1.003001",
    "ExtUtils::MakeMaker" => 0,
    "File::ShareDir" => 0,
    "File::Spec" => 0,
    "Test::DZil" => 0,
    "Test::More" => 0
  },
  "VERSION" => "1.001003",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::Plugin::Bootstrap::lib" => "0.04000000",
  "Dist::Zilla::Role::Bootstrap" => 0,
  "Dist::Zilla::Util::Test::KENTNL" => "1.003001",
  "ExtUtils::MakeMaker" => 0,
  "File::ShareDir" => 0,
  "File::Spec" => 0,
  "Moose" => 0,
  "Path::Tiny" => 0,
  "Test::DZil" => 0,
  "Test::File::ShareDir::Object::Module" => 0,
  "Test::More" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4206"
  },
  "configure_requires" => {
    "Module::Build" => "0.4206"
  },
  "dist_abstract" => "Use a C<share> directory on your dist for a module during bootstrap",
  "dist_author" => [
    "Kent Fredric <kentfredric\@gmail.com>"
  ],
  "dist_name" => "Dist-Zilla-Plugin-Bootstrap-ShareDir-Module",
  "dist_version" => "0.2.1",
  "license" => "perl",
  "module_name" => "Dist::Zilla::Plugin::Bootstrap::ShareDir::Module",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Dist::Zilla::Role::Bootstrap" => 0,
    "Moose" => 0,
    "MooseX::AttributeShortcuts" => 0,
    "Path::Tiny" => 0,
    "Test::File::ShareDir::Object::Module" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [],
  "test_requires" => {
    "Cwd" => 0,
    "Dist::Zilla::Plugin::Bootstrap::lib" => "0.04000000",
    "ExtUtils::MakeMaker" => 0,
    "File::Copy::Recursive" => 0,
    "File::ShareDir" => 0,
    "File::Spec" => 0,
    "Path::FindDev" => 0,
    "Test::DZil" => 0,
    "Test::Fatal" => 0,
    "Test::More" => "1.001003"
  }
);


my %fallback_build_requires = (
  "Cwd" => 0,
  "Dist::Zilla::Plugin::Bootstrap::lib" => "0.04000000",
  "ExtUtils::MakeMaker" => 0,
  "File::Copy::Recursive" => 0,
  "File::ShareDir" => 0,
  "File::Spec" => 0,
  "Module::Build" => "0.4206",
  "Path::FindDev" => 0,
  "Test::DZil" => 0,
  "Test::Fatal" => 0,
  "Test::More" => "1.001003"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
